test/cets_SUITE.erl

```diff
@@ -0,0 +1,359 @@
+
+insert_many_with_one_record(_Config) ->
+    cets:start(ins1m, #{}),
+    cets:insert_many(ins1m, [{alice, 32}]),
+    [{alice, 32}] = ets:lookup(ins1m, alice).
+
+insert_many_with_two_records(_Config) ->
+    cets:start(ins2m, #{}),
+    cets:insert_many(ins2m, [{alice, 32}, {bob, 55}]),
+    [{alice, 32}, {bob, 55}] = ets:tab2list(ins2m).
+
+delete_works(_Config) ->
+    cets:start(del1, #{}),
+    cets:insert(del1, {alice, 32}),
+    cets:delete(del1, alice),
+    [] = ets:lookup(del1, alice).
+
+delete_many_works(_Config) ->
+    cets:start(del1, #{}),
+    cets:insert(del1, {alice, 32}),
+    cets:delete_many(del1, [alice]),
+    [] = ets:lookup(del1, alice).
+
+join_works(_Config) ->
+    {ok, Pid1} = cets:start(join1tab, #{}),
+    {ok, Pid2} = cets:start(join2tab, #{}),
+    ok = cets_join:join(join_lock1, #{}, Pid1, Pid2).
+
+inserted_records_could_be_read_back_from_replicated_table(_Config) ->
+    {ok, Pid1} = cets:start(ins1tab, #{}),
+    {ok, Pid2} = cets:start(ins2tab, #{}),
+    ok = cets_join:join(join_lock1_ins, #{}, Pid1, Pid2),
+    cets:insert(ins1tab, {alice, 32}),
+    [{alice, 32}] = ets:lookup(ins2tab, alice).
+
+join_works_with_existing_data(_Config) ->
+    {ok, Pid1} = cets:start(ex1tab, #{}),
+    {ok, Pid2} = cets:start(ex2tab, #{}),
+    cets:insert(ex1tab, {alice, 32}),
+    %% Join will copy and merge existing tables
+    ok = cets_join:join(join_lock1_ex, #{}, Pid1, Pid2),
+    [{alice, 32}] = ets:lookup(ex2tab, alice).
+
+%% This testcase tests an edgecase: inserting with the same key from two nodes.
+%% Usually, inserting with the same key from two different nodes is not possible
+%% (because the node-name is a part of the key).
+join_works_with_existing_data_with_conflicts(_Config) ->
+    {ok, Pid1} = cets:start(con1tab, #{}),
+    {ok, Pid2} = cets:start(con2tab, #{}),
+    cets:insert(con1tab, {alice, 32}),
+    cets:insert(con2tab, {alice, 33}),
+    %% Join will copy and merge existing tables
+    ok = cets_join:join(join_lock1_con, #{}, Pid1, Pid2),
+    %% We insert data from other table into our table when merging, so the values get swapped
+    [{alice, 33}] = ets:lookup(con1tab, alice),
+    [{alice, 32}] = ets:lookup(con2tab, alice).
+
+join_with_the_same_pid(_Config) ->
+    {ok, Pid} = cets:start(joinsame, #{}),
+    %% Just insert something into a table to check later the size
+    cets:insert(joinsame, {1, 1}),
+    link(Pid),
+    ok = cets_join:join(joinsame_lock1_con, #{}, Pid, Pid),
+    Nodes = [node()],
+    %% The process is still running and no data loss (i.e. size is not zero)
+    #{nodes := Nodes, size := 1} = cets:info(Pid).
+
+test_multinode(Config) ->
+    Node1 = node(),
+    [Node2, Node3, Node4] = proplists:get_value(nodes, Config),
+    Tab = tab1,
+    {ok, Pid1} = start(Node1, Tab),
+    {ok, Pid2} = start(Node2, Tab),
+    {ok, Pid3} = start(Node3, Tab),
+    {ok, Pid4} = start(Node4, Tab),
+    ok = join(Node1, Tab, Pid3, Pid1),
```

Comment:
This happens a few times: the first pid is remote, and the second is local, while in the `join` function it is the other way around. Maybe don't swap the order to make it easier to understand?